swagger/swagger-generator/generator-core/src/test/java/org/apache/servicecomb/swagger/generator/core/TestParamUtils.java

```diff
@@ -95,4 +104,39 @@ public void setParameterTypeByTypeOnComplexType() {
           + "param name is [testName]", e.getMessage());
     }
   }
+
+  private static class AllTypeTest {
+    TestType1 t1;
+
+    TestType2 t2;
+
+    List<TestType1> t3;
+
+    Map<String, TestType2> t4;
+
+    TestType2[] t5;
+  }
+
+
+  @Test
+  public void testAddDefinitions() {
+    Field[] fields = AllTypeTest.class.getDeclaredFields();
+    for (int i = 0; i < fields.length; i++) {
+      for (int j = 0; j < fields.length; j++) {
+        if (i != j) {
+          try {
+            testExcep(fields[i].getGenericType(), fields[j].getGenericType());
```

Comment:
Hi, here you can add a `fail` assertion here to ensure there must be an Exception. Like below:
```java
    try {
      PojoService.test.testException(557);
      fail("InvocationException expected, but threw nothing");
    } catch (InvocationException e) {
      assertThat(e.getStatusCode(), is(557));
      assertThat(e.getErrorData().toString(), is("[[557 error]]"));
    } catch (Exception e) {
      fail("InvocationException expected, but threw " + e);
    }
```